torch/_inductor/graph.py

```diff
@@ -285,11 +285,15 @@ def constant_name(self, name: str, device_override: torch.device):
         return alt_name
 
     def placeholder(self, target: str, args, kwargs):
-        example: torch.Tensor = super().placeholder(target, args, kwargs)
+        example = super().placeholder(target, args, kwargs)
         if isinstance(example, SymTypes):
             expr = example.node.expr
             self.graph_inputs[target] = expr
             return expr
+        elif isinstance(example, int):
+            expr = sympy.Integer(example)
+            self.graph_inputs[target] = expr
+            return expr
```

Comment:
We still need to handle `int` here despite having support for `SymTypes`.

SymPy Mul is printed as `*`, so the result is an `int` when both args are concrete. See `32*s0` below, which is eval'd to `256`.

```
pytest -rfxs test/inductor/test_torchinductor_dynamic_shapes.py -k test_lowmem_dropout2_dynamic_shapes_cpu -v --capture=no


[cpp code removed]

def call(args):
    primals_1, primals_2, primals_3 = args
[...]
    s0 = primals_3_size[0]
[...]
    return (buf3, primals_3, seed_cpu_None.clone(), buf1, as_strided(primals_2, (32, 32), (32, 1)), 32*s0, )


def benchmark_compiled_module():
[...]
    primals_3 = rand_strided((8, 32), (32, 1), device='cpu', dtype=torch.float32)
    print_performance(lambda: call([primals_1, primals_2, primals_3]))
```

Reply:
I would quite like to understand this situation, because fixing this will solve a lot of models. However, I don't quite understand your explanation here. The example value here is passed while we are doing define-by-run lowering. Why is the pasted code (which is the generated benchmark code) relevant here?

Reply:
Looks like this `int` block became unnecessary anyway after the last rebase. I've removed it in a separate commit to confirm: https://github.com/pytorch/pytorch/pull/96230

Reply:
So do you still hard-code the number of elements for backward, or has this been fixed too?

Reply:
> So do you still hard-code the number of elements for backward, or has this been fixed too?

It's been fixed. The constants in kernels have been replaced by muls. See here https://gist.github.com/nkaretnikov/25bbe1375f582c754055f00f42346408

```
git diff --no-index --word-diff lowmem.txt dyn_lowmem.txt
git diff --no-index --word-diff rand.txt dyn_rand.txt
```